Update-AutomationAzureModulesForAccount.ps1

```diff
@@ -107,41 +110,26 @@ function ConvertJsonDictTo-HashTable($JsonString) {
 # Use the Run As connection to login to Azure
 function Login-AzureAutomation([bool] $AzModuleOnly) {
     try {
-        $RunAsConnection = Get-AutomationConnection -Name "AzureRunAsConnection"
         Write-Output "Logging in to Azure ($AzureEnvironment)..."
         
-        if (!$RunAsConnection.ApplicationId) {
-            $ErrorMessage = "Connection 'AzureRunAsConnection' is incompatible type."
-            throw $ErrorMessage            
-        }
-        
         if ($AzModuleOnly) {
-            Connect-AzAccount `
-                -ServicePrincipal `
-                -TenantId $RunAsConnection.TenantId `
-                -ApplicationId $RunAsConnection.ApplicationId `
-                -CertificateThumbprint $RunAsConnection.CertificateThumbprint `
-                -Environment $AzureEnvironment
-
-            Select-AzSubscription -SubscriptionId $RunAsConnection.SubscriptionID  | Write-Verbose
+            # Ensures you do not inherit an AzContext in your runbook
+            Disable-AzContextAutosave -Scope Process
+            # Connect to Azure with system-assigned managed identity. 
+            # Please enable appropriate RBAC permissions to the system identity of this automation account. Otherwise, the runbook may fail...
+            $context = (Connect-AzAccount -Identity -Environment $AzureEnvironment).Context
+
+            # set and store context
+            $AzureContext = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+            Select-AzSubscription -SubscriptionId $SubscriptionId  | Write-Verbose
         } else {
-            Add-AzureRmAccount `
-                -ServicePrincipal `
-                -TenantId $RunAsConnection.TenantId `
-                -ApplicationId $RunAsConnection.ApplicationId `
-                -CertificateThumbprint $RunAsConnection.CertificateThumbprint `
-                -Environment $AzureEnvironment
-
-            Select-AzureRmSubscription -SubscriptionId $RunAsConnection.SubscriptionID  | Write-Verbose
+            # Connect to Azure with system-assigned managed identity. 
+            # Please enable appropriate RBAC permissions to the system identity of this automation account. Otherwise, the runbook may fail...
+            $context = (Connect-AzureRmAccount -Identity -Environment $AzureEnvironment).Context
+            Set-AzureRmContext -SubscriptionId $SubscriptionId -ErrorAction Stop

```

Comment:
We should capture the context here as we have potentially changed the subscription

$context = Set-AzureRmContext -SubscriptionId $SubscriptionId -ErrorAction Stop